### DISCUSSION TITLE
Boundary term in ADDiffusion tutorial

### DISCUSSION THREAD

**Discussion by vishankkumar** [idaholab/moose#16322] [Q&A Modules: General] [answered]
I was going through again the tutorial of ADDiffusion with steady state solution and I had a question regarding the boundary term. It was also discussed in the MOOSE workshop2020. When you write down the weak form of laplacian, you get a symmetric matrix term of _grad_u[_qp] * _grad_test[_qp] and a Newmann boundary term of _grad_u[_qp] * _test_[qp] * n. However, in the input file you take the Dirichlet BC term, so where is this term get adjusted or am I missing something?

Thanks for your response.

**Comment by hugary1995** [comment, marked as answer]
Are you referring to Eq. 5 from [this page](https://mooseframework.inl.gov/getting_started/examples_and_tutorials/tutorial01_app_development/step04_weak_form.html#residuals)? The bold statement below "**This is the weak form of the Laplace equation**" is wrong.
Eq. 5 is what you get after applying divergence theorem on Eq. 2, but it is not the weak form of the strong form.

I happen to have a simple write-up at hand:

![image](https://user-images.githubusercontent.com/31555204/100543649-56688f00-321f-11eb-8fe0-c78d96f1b48d.png)

As you can see, Dirichlet BCs are involved in the construction of proper test and trial function spaces. They are constraints on certain DoFs a priori, and they are not weakly satisfied. Therefore, you don't see Dirichlet BCs in the weak form.

**Reply by hugary1995** [reply]
The bold statement probably need some rephrasing.

That being said, I still think the tutorial on the website is phenomenal -- it has done a remarkable job at explaining the workflow of MOOSE development. It is always difficult to find a balance between "writing a short and concise tutorial" and "explaining the nitty-gritty details of everything".

**Reply by vishankkumar** [reply]
Thank you @hugary1995 for the quick explanation. So Dirichlet BC has been used for constructing the test and the trial functions, which we define in the input file. However, my concern was more on the boundary term, $\int_{\Gamma_n}whdA$. This $h$ is the Newmann boundary condition, if I am not wrong so does it has to be defined in the BC or as an additional term in the kernel or it is within preComputeQPResidual?

And I understand, its always difficult to put things crisp and crystal clear for everyone but the documentation of moose has done a good job.

**Reply by hugary1995** [reply]
If h is nonzero, then yes, it has to be defined using a BC object, e.g. ADNeumannBC, ADFunctionNeumannBC, etc..

If h is zero then you don't need to do anything. It is clear from the weak form that if you drop the boundary term the Neumann BC will be weakly zero.
